crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -86,16 +86,16 @@ pub struct Immutable {
 }
 
 /// A shared object is an object that is shared using the 0x2::transfer::share_object function
-/// and is accessible to everyone.
-/// Unlike owned objects, once an object is shared, it stays mutable and can be accessed by anyone,
-/// unless it is made immutable. An example of immutable shared objects are all published packages
-/// and modules on Sui.
+/// and is accessible to everyone. There are two types of shared objects: mutable and immutable.
+/// Unlike owned objects, once an object is shared, it stays mutable and can be accessed by anyone.
+/// An example of immutable shared objects are all published packages and modules on Sui.
 #[derive(SimpleObject, Clone)]
 pub struct Shared {
     initial_shared_version: u64,
 }
 
-/// The parent of this object
+/// The parent of this object.
+/// If the object's owner is a Parent, this object is part of a dynamic field (it is the value of the dynamic field, or the intermediate Field object itself).
```

Comment:
nit: We could probably get rid of the first line now?  It's also maybe worth calling out that if the owner is a Parent, we know that that Parent must be another Object (unlike in the `Address` case where it could be either).